airflow/models/mappedoperator.py

```diff
@@ -569,6 +573,7 @@ def _get_map_lengths(self, run_id: str, *, session: Session) -> Dict[str, int]:
                 map_lengths[mapped_arg_name] += length
         return map_lengths
 
+    @cache
     def _resolve_map_lengths(self, run_id: str, *, session: Session) -> Dict[str, int]:
```

Comment:
It’ll be difficult to actually hit this cache due to `session` :(

Reply:
Session has a longer life time than you might expect due to SQLA's Pooling

Reply:
```python-console
In [2]: @provide_session
   ...: def a(session):
   ...:     print(id(session))
   ...:     b()

In [4]: @provide_session
   ...: def b(session):
   ...:     print(id(session))

In [5]: a()
140533469520240
140533469520240

```